baseapp/baseapp_test.go

```diff
@@ -903,6 +911,14 @@ func handlerMsgCounter(t *testing.T, capKey storetypes.StoreKey, deliverKey []by
 		}
 
 		res.Events = ctx.EventManager().Events().ToABCIEvents()
+
+		any, err := codectypes.NewAnyWithValue(msg)
+		if err != nil {
+			return nil, err
+		}
+
+		res.MsgResponses = []*codectypes.Any{any}
```

Comment:
Doing all this msgCounter stuff here in baseapp_test is a bit hacky IMO. Let's try to refactor that out in #10282